sdk/typescript/src/signers/signer.ts

```diff
@@ -31,4 +36,61 @@ export abstract class Signer {
    * Returns the signature for the data and the public key of the signer
    */
   abstract signData(data: Base64DataBuffer): Promise<SignaturePubkeyPair>;
+
+  // Returns a new instance of the Signer, connected to provider.
+  // This MAY throw if changing providers is not supported.
+  abstract connect(provider: Provider): Signer;
+
+  ///////////////////
+  // Sub-classes MAY override these
+
+  /**
+   * Sign a transaction and submit to the Gateway for execution
+   *
+   * @param txBytes BCS serialised TransactionData bytes
+   */
+  async signAndExecuteTransaction(
+    txBytes: Base64DataBuffer
+  ): Promise<TransactionResponse> {
+    this._checkProvider('signAndExecuteTransaction');
+    const sig = await this.signData(txBytes);
+    return await this.provider!.executeTransaction({
```

Comment:
i've never seen this `provider!.execute()` syntax before - what is the `!` doing ?

Reply:
That's the non-null assertion operator. It is a way to tell the compiler "this expression cannot be null or undefined here, so don't complain about the possibility of it being null or undefined." 

https://stackoverflow.com/questions/42273853/in-typescript-what-is-the-exclamation-mark-bang-operator-when-dereferenci